File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer =[]
    
    if yellow == 1: 
        return [3,3]
    
    for i in range(1,yellow):
        if yellow % i == 0 and brown == 4 + 2*i + 2*(yellow/i):
            answer = [yellow/i+2,i+2]
            
    answer.sort(reverse=True)
    return answer

# yellow = a*b
# brown = 4 + 2a + 2b

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const size = brown+yellow;
    let answer = [];
    for (i = 3; i <= size/3; i++){
        if (size % i === 0){
            if ((size/i -2)*(i-2) === yellow){
                answer = [i,size/i];
            }
        }
    }
    answer.sort((a,b)=>b-a);
    return answer;
}
